### PR TITLE
Add client code and use cases for cloning and uploading source code

### PR DESCRIPTION
**Pull request by herman-rogers**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -4,17 +4,49 @@ import (
...
+	if os.Getenv(config.GoEnv) == "development" {
+		mockdata := `{"id":"123456","usr":"Boomer","repo":"https://github.com/Gamebuildr/Gogeta.git","proj":"Gogeta","type":"git"}`
+		mockMessages := testutils.StubbedQueueMessage(mockdata)
+		gogeta.Queue = queuesystem.AmazonQueue{
+			Client: testutils.MockedAmazonClient{Response: mockMessages.Resp},
+			URL:    "mockUrl_%d",
+		}
+	}
+	runQueuePoll(&gogeta)
+	//gocron.Every(1).Minute().Do(runQueuePoll)
```
Comment (herman-rogers) [inline main.go:30]: still have yet to implement the poller and the server process to keep alive.

File: pkg/messages/messenger.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package messages
+
```
Comment (herman-rogers) [inline pkg/messages/messenger.go:2]: Still need to implement a messaging system that can send messages up to SNS for MrRobot and Jambuildr

Comment (herman-rogers) [reply pkg/messages/messenger.go:2]: Jambuildr being out pivot to building an assortment of Jams - have yet to implement raspberry

File: pkg/queuesystem/queue_system.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+// Messages is the base struct for getting
+// data from a specified queue system
+type Messages interface {
+	GetQueueMessages() ([]QueueMessage, error)
+	DeleteMessageFromQueue(string) error
+}
+
+// QueueMessage is the abstraction for formatting
+// and using data that comes from the queues
+type QueueMessage struct {
```
Comment (herman-rogers) [inline pkg/queuesystem/queue_system.go:12]: Need your input on what data Gogeta should store from Gamebuildr to pass onto Mr.Robot

File: main.go [hunk 4]
```diff
@@ -4,17 +4,49 @@ import (
...
 	//startServer()
 }
 
+func runQueuePoll(gogeta *client.GogetaClient) {
+	repo := gogeta.GetSourceCode()
+
+	if repo.SourceLocation != "" {
+		archive := os.Getenv("GOPATH") + "/repos/" + repo.ProjectName + ".zip"
```
Comment (chvck) [inline main.go]: May want to use path.Join here

File: pkg/queuesystem/queue_system.go (hunk 3 above)
Comment (chvck) [reply pkg/queuesystem/queue_system.go:12]: type StandardBuildRequest struct {
	RepoFolder     string `json:"repofolder"`
	ConfigPath     string `json:"configpath"`
	EngineName     string `json:"enginename"`
	EnginePlatform string `json:"engineplatform"`
	EngineVersion  string `json:"engineversion"`
	Success        bool   `json:"success"`
	Log            string `json:"log"`
	BuildrId       string `json:"buildrid"`
	BuildId        string `json:"buildid"`
	Type           string `json:"type"`
}

Could probably do with ProjectName in there too

Comment (chvck) [reply pkg/queuesystem/queue_system.go:12]: So, probably don't need ConfigPath, Success, Log, Type.

File: pkg/sourcesystem/source_interface.go [hunk 5]
```diff
@@ -48,7 +47,6 @@ func (scm SystemSCM) UpdateSource(repo SourceRepository) error {
 
 func createSourceFolder(project string) string {
 	uuid := uuid.NewV4()
-	folderName := project + "_" + uuid.String()
-	repoPath := os.Getenv(config.RepoPath)
-	return repoPath + folderName
+	folderName := "/repos/" + project + "_" + uuid.String()
```
Comment (chvck) [inline pkg/sourcesystem/source_interface.go]: path.Join maybe

File: pkg/storehouse/zip_compression.go [hunk 6]
```diff
@@ -10,18 +10,14 @@ import (
 
 // Zip compression will take a file or
 // directory and create a zip archive
-type Zip struct {
-	Source string
-	Target string
-}
+type Zip struct{}
```
Comment (chvck) [inline pkg/storehouse/zip_compression.go:13]: Do we wanna move this out to its own go package?

File: pkg/sourcesystem/source_interface.go (hunk 5 above)
Comment (herman-rogers) [reply pkg/sourcesystem/source_interface.go]: Will add these to these two bits

File: pkg/storehouse/zip_compression.go (hunk 6 above)
Comment (herman-rogers) [reply pkg/storehouse/zip_compression.go:13]: Could do? Probably a good idea to - easy enough to extract out.

Comment (chvck) [reply pkg/storehouse/zip_compression.go:13]: Well, I need to zip stuff too

**Pull request merged**